sympy/core/expr.py

```diff
@@ -332,7 +332,7 @@ def __ge__(self, other):
         except SympifyError:
             raise TypeError("Invalid comparison %s >= %s" % (self, other))
         for me in (self, other):
-            if me.is_complex and me.is_extended_real is False:
+            if (me.is_complex or me.is_infinite) and me.is_extended_real is False:
```

Comment:
In general this replacement will not be equivalent since if `x` is complex then `(1/x)` will be complex or infinite but `(1/x).is_complex` and `(1/x).is_infinite` will both return None. It isn't possible to make something equivalent without having extended_complex.

Reply:
Do we need the `is_complex` check here at all? Shouldn't we just raise if extended_real is False?

Reply:
(Same goes for the other comparison methods below.)